global_run.py

```diff
@@ -0,0 +1,82 @@
+import random
+from copy import deepcopy
+import pickle as pkl
+from tqdm import tqdm
+import os
+import time
+
+
+def subsample(samples, n=1000):
+    selected_idxes = list(range(len(samples)))
+    random.shuffle(selected_idxes)
+    selected_idxes = selected_idxes[:n]
+    return [samples[i] for i in sorted(selected_idxes)]
+
+# flip the problem of describing corpus A to describing corpus B
+def flip_problem(problem):
+    problem = deepcopy(problem)
+    problem['A_desc'], problem['B_desc'] = problem['B_desc'], problem['A_desc']
+    problem['split'] = {
+        k: {
+            'A_samples': v['B_samples'],
+            'B_samples': v['A_samples']
+        } for k, v in problem['split'].items()
+    }
+    return problem
+
+
+if __name__ == '__main__':
+    from D5 import D5
+    from validator import DummyValidator, Validator
+    from lm_proposer import GPT3_Proposer
+    from get_representative import return_extreme_values
+
+    problems = pkl.load(open('OpenD5.pkl', 'rb'))
+
+    # the default validator has 11B parameters
+    validator = Validator()
+    if not os.path.exists('discoveries'):
+        os.mkdir('discoveries')
+
+    # randomly shuffle the problems to run our system on
+    problem_idxes = list(range(len(problems)))
+    random.shuffle(problem_idxes)
+    pbar = tqdm(problem_idxes)
+
+
+    for problem_id in pbar:
+        pbar.set_description(f'problem {problem_id}')
+        problem_orig = problems[problem_id]
+
+        def get_h2h_dicts(problem, save_path):
+            current_time = time.time()
+            if os.path.exists(save_path):
+                loaded_object = pkl.load(open(save_path, 'rb'))
```

Comment:
You should use `with open(save_path, 'rb'): ...` since it makes sure that the file gets closed if an exception occurs.

Reply:
As a follow-up you probably should not be pickling this object at all. It should be saved using pythons built in `json` library see https://docs.python.org/3/library/json.html.